### DISCUSSION TITLE
Impossible to install EfficientWord-Net v1.0.2 on RPi 4 with Raspbian 10 (BUSTER)

### DISCUSSION THREAD

**Discussion by amikosolucoes-p** [Ant-Brain/EfficientWord-Net#32] [Q&A]
When trying to install the `EfficientWord-Net v1.0.2` lib on the Raspberry Pi 4, an error related to `onnxruntime 1.14.1` displays the following error:
```
Could not find a version that satisfies the requirement onnxruntime (from versions: )
No matching distribution found for onnxruntime
```

Apparently my OS architecture is not compatible with this lib.
```
PRETTY_NAME="Raspbian GNU/Linux 10 (buster)" 
NAME="Raspbian GNU/Linux"  
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"
```

Has anyone had a similar situation and managed to resolve it?

_NOTE: It is not possible to change the OS or Raspberry._

**Comment by lucca-amiko** [comment]
I've the same problem too.

**Comment by TheSeriousProgrammer** [comment]
The latest changes have not been tested in a raspberrypi , but we should be able to run onnx models in a pi, let me have a look at it this weekend

**Reply by amikosolucoes-p** [reply]
Thanks! If you need more information about the problem, I am available

**Comment by TheSeriousProgrammer** [comment]
I have found the issue
Onnxruntime binaries are available only for 64bit ARM operating systems, its possible that you could be running a 32 bit version of the same. (Raspberry pi's official os has a 64 bit variant as well hidden inside the website. Its a beta version and I use that). Moreover the binaries available for python3.8+. I'll try to build a arm7l binary for 32bit os. Or you can compile the binary using the ones provided in the website

**Reply by TheSeriousProgrammer** [reply]
You can see the same in the release page of onnxruntime website https://pypi.org/project/onnxruntime/#files
